sample/get-bucket.c

```diff
@@ -5,51 +5,34 @@
 #include <sys/types.h>
 #include <sys/stat.h>
 #include "ds3.h"
+#include "samples.h"
 
 int main(void) {
 
-    ///*
-    // Get Bucket
-
+   // Get Bucket
+    ds3_client* client;
+    ds3_request* request;
+    ds3_error* error;
     ds3_get_bucket_response *response;
     uint64_t i;
 
-    // Setup client credentials and then the actual client itself.
-    ds3_creds * creds = ds3_create_creds("cnlhbg==","ZIjGDQAs");
-    ds3_client * client = ds3_create_client("http://192.168.56.101:8080", creds);
-
-    // You can optionally set a proxy server that a request should be sent through
-    //ds3_client_proxy(client, "192.168.56.1:8888");
-
-    char * bucket = "books";
+    // Create a client from environment variables
+    error = ds3_create_client_from_env(&client);
+    handle_error(error);
 
     // Create the get bucket request.
-    ds3_request* request = ds3_init_get_bucket(bucket);
+    request = ds3_init_get_bucket(BUCKETNAME);
 
     // This performs the request to a DS3 appliance.
     // If there is an error 'error' will not be NULL
     // If the request completed successfully then 'error' will be NULL
-    ds3_error* error = ds3_get_bucket(client, request, &response);
+    error = ds3_get_bucket(client, request, &response);
     ds3_free_request(request);
 
-    // Check that the request completed successfully
-    if(error != NULL) {
-        if(error->error != NULL) {
-            printf("Got an error (%lu): %s\n", error->error->status_code, error->message->value);
-            printf("Message Body: %s\n", error->error->error_body->value);
-        }
-        else {
-            printf("Got a runtime error: %s\n", error->message->value);
-        }
-        ds3_free_error(error);
-        ds3_free_creds(creds);
-        ds3_free_client(client);
-        return 1;
-    }
-
+	handle_error(error);
```

Comment:
What is it supposed to be?

jk

From: Ryan Moore [mailto:notifications@github.com]
Sent: Wednesday, December 09, 2015 4:51 PM
To: SpectraLogic/ds3_c_sdk <ds3_c_sdk@noreply.github.com>
Cc: John Kranz <johnk@spectralogic.com>
Subject: Re: [ds3_c_sdk] Build in Debian 8 (CSDK-76) (#83)


In sample/get-bucket.c<https://github.com/SpectraLogic/ds3_c_sdk/pull/83#discussion_r47171369>:

> -    // Check that the request completed successfully

> -    if(error != NULL) {

> -        if(error->error != NULL) {

> -            printf("Got an error (%lu): %s\n", error->error->status_code, error->message->value);

> -            printf("Message Body: %s\n", error->error->error_body->value);

> -        }

> -        else {

> -            printf("Got a runtime error: %s\n", error->message->value);

> -        }

> -        ds3_free_error(error);

> -        ds3_free_creds(creds);

> -        ds3_free_client(client);

> -        return 1;

> -    }

> -

> +     handle_error(error);

The tabbing looks a bit off here.

—
Reply to this email directly or view it on GitHub<https://github.com/SpectraLogic/ds3_c_sdk/pull/83/files#r47171369>.